src/Catalyst.Core.Lib.Tests/UnitTests/Rpc/IO/Observers/PeerCountRequestObserverTests.cs

```diff
@@ -0,0 +1,124 @@
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Catalyst.Abstractions.IO.Messaging.Dto;
+using Catalyst.Core.Lib.Extensions;
+using Catalyst.Core.Lib.Network;
+using Catalyst.Core.Lib.P2P.Models;
+using Catalyst.Core.Lib.P2P.Repository;
+using Catalyst.Core.Modules.Rpc.Server.IO.Observers;
+using Catalyst.Protocol.Wire;
+using Catalyst.Protocol.Rpc.Node;
+using Catalyst.TestUtils;
+using DotNetty.Transport.Channels;
+using FluentAssertions;
+using Microsoft.Reactive.Testing;
+using NSubstitute;
+using Serilog;
+using Xunit;
+
+namespace Catalyst.Core.Lib.Tests.UnitTests.Rpc.IO.Observers
+{
+    /// <summary>
+    ///     Tests the peer count CLI and RPC calls
+    /// </summary>
+    public sealed class PeerCountRequestObserverTests
+    {
+        private readonly TestScheduler _testScheduler;
+
+        /// <summary>The logger</summary>
+        private readonly ILogger _logger;
+
+        /// <summary>The fake channel context</summary>
+        private readonly IChannelHandlerContext _fakeContext;
+
+        /// <summary>
+        ///     Initializes a new instance of the
+        ///     <see>
+        ///         <cref>PeerListRequestObserverTest</cref>
+        ///     </see>
+        ///     class.
+        /// </summary>
+        public PeerCountRequestObserverTests()
+        {
+            _testScheduler = new TestScheduler();
+            _logger = Substitute.For<ILogger>();
+            _fakeContext = Substitute.For<IChannelHandlerContext>();
+            var fakeChannel = Substitute.For<IChannel>();
+            _fakeContext.Channel.Returns(fakeChannel);
+        }
+
+        /// <summary>
+        ///     Tests the peer count request and response.
+        /// </summary>
+        /// <param name="fakePeers">The peer count.</param>
+        [Theory]
+        [InlineData(40)]
+        [InlineData(20)]
+        public void TestPeerListRequestResponse(int fakePeers)
+        {
+            var peerRepository = Substitute.For<IPeerRepository>();
+            var peerList = new List<Peer>();
+
+            for (var i = 0; i < fakePeers; i++)
+            {
+                peerList.Add(new Peer
+                {
+                    Reputation = 0,
+                    LastSeen = DateTime.Now,
+                    PeerId = PeerIdHelper.GetPeerId(i.ToString())
+                });
+            }
+
+            // Build a fake remote endpoint
+            _fakeContext.Channel.RemoteAddress.Returns(EndpointBuilder.BuildNewEndPoint("192.0.0.1", 42042));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make sure using this hardcoded IP address '192.0.0.1' is safe here.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)